test/built-ins/Object/getOwnPropertyDescriptors/duplicate-keys.js

```diff
@@ -21,7 +21,7 @@ var proxy = new Proxy({}, {
   },
   getOwnPropertyDescriptor(t, name) {
     log += 'getOwnPropertyDescriptor:' + name + '|';
-    return descriptors[i++];
+    return descriptors[i >= descriptors.length ? descriptors.length - 1 : i++];
```

Comment:
no, its the assertion on line  30 that is wrong here, it shouldn't be overwritten with undefined.